pkg/registry/rbac/clusterrolebinding/policybased/storage.go

```diff
@@ -81,7 +82,7 @@ func (s *Storage) Create(ctx context.Context, obj runtime.Object, createValidati
 	if err != nil {
 		return nil, err
 	}
-	rules, err := s.ruleResolver.GetRoleReferenceRules(v1RoleRef, metav1.NamespaceNone)
+	rules, err := s.ruleResolver.GetRoleReferenceRules(genericapirequest.WithNamespace(ctx, ""), v1RoleRef, metav1.NamespaceNone)
```

Comment:
This is wrapping the context, and inside DefaultRuleResolver#GetRoleReferenceRules we're wrapping the context again. Drop this wrapping and let the wrapping happen when we pass a context to a registry (a good clue for when we do that is when we inject constructed `GetOptions{}` or `ListOptions{}` or where we were passing `genericapirequest.NewContext()` before)